src/components/select/bl-select.ts

```diff
@@ -241,58 +330,74 @@ export default class BlSelect extends LitElement {
       'select-wrapper': true,
       'select-open': this.opened,
       'selected': this._selectedOptions.length > 0,
-      'invalid': this._isInvalid,
+      'invalid': !this.validity.valid,
+      'dirty': this.dirty
     })}
-      tabindex="-1"
+      @keydown=${this.handleKeydown}
     >
-      ${label} ${this.inputTemplate()} ${this.menuTemplate()} ${helpMessage} ${invalidMessage}
+      ${label}
+      ${this.inputTemplate()}
+      <div class="popover" tabindex="${ifDefined(this._isPopoverOpen ? undefined : '-1')}" @bl-select-option=${this._handleSelectOptionEvent}>
+        <slot></slot>
+      </div>
+      <div class="hint">${invalidMessage} ${helpMessage}</div>
     </div> `;
   }
 
-  private get _showPlaceHolder() {
-    if (this.label && !this.labelFixed) {
-      return !this._selectedOptions.length && this._isPopoverOpen;
+  private focusedOptionIndex = -1;
+
+  private handleKeydown(event: KeyboardEvent) {
+    if (this.focusedOptionIndex === -1 && event.code === 'Enter' || event.code === 'Space') {
+      this.togglePopover();
+    } else if (event.code === 'Escape') {
+      this.close();
+      event.preventDefault();
+    } else if (this._isPopoverOpen && ['ArrowDown', 'ArrowUp'].includes(event.code)) {
+      event.code === 'ArrowDown' && this.focusedOptionIndex++;
```

Comment:
I noticed one thing here that when you press arrow keys it also scrolls the page along with the option navigation. Is this expected? somehow it gives inconsistent effect to the users. maybe shall we `event.preventDefault()` and then only navigate the options?

Reply:
Also, after pressing the `Space` it scrolls me down of the page, I think we should preventDefault there too.

Reply:
I also noticed this but somehow found it logical too. I think browser is keeping focused element in screen by default and if it's in a scrollable area, scrolls the container to focused element. Select options can be a long list and if we try to avoid this behaviour, we would need to handle scrolling the container to focused element if it's outside of the visible area, manually. So I'm not sure if it worths to do that.

> Also, after pressing the Space it scrolls me down of the page, I think we should preventDefault there too.

I am going to check this one.

Reply:
Fixed the scrolling issue. Preventing the default behavior helped.